src/coreclr/jit/promotion.cpp

```diff
@@ -436,6 +446,15 @@ class LocalUses
         // TODO-CQ: A store-forwarding optimization in lowering could get rid
         // of these copies; however, it requires lowering to be able to prove
         // that not writing the fields into the struct local is ok.
+        //
+        // Note: Technically we also introduce writebacks before returns that
+        // we could account for, however the returns we see during physical
+        // promotion are only for structs returned in registers and in most
+        // cases the writeback introduced is just replacing an existing
+        // "natural" writeback so it ends up being zero cost.
```

Comment:
perhaps change to "zero additional cost" indicating that it is balanced not that the new allocation is free?

Reply:
Good idea! Reworded it to make it more clear.